packages/flutter_tools/lib/src/ios/migrations/uiapplicationmain_deprecation_migration.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+const String _appDelegateFileBefore = r'''
+@UIApplicationMain
+@objc class AppDelegate''';
+
+const String _appDelegateFileAfter = r'''
+@main
+@objc class AppDelegate''';
+
+/// Replace the deprecated `@UIApplicationMain` attribute with `@main`.
+///
+/// See:
+/// https://github.com/apple/swift-evolution/blob/main/proposals/0383-deprecate-uiapplicationmain-and-nsapplicationmain.md
+class UIApplicationMainDeprecationMigration extends ProjectMigrator {
+  UIApplicationMainDeprecationMigration(
+    IosProject project,
+    super.logger,
+  ) : _appDelegateSwift = project.appDelegateSwift;
+
+  final File _appDelegateSwift;
+
+  @override
+  Future<bool> migrate() async {
+    // Skip this migration if the project uses Objective-C.
+    if (!_appDelegateSwift.existsSync()) {
+      return true;
```

Comment:
I think we usually `trace` here:
```suggestion
      logger.printTrace('AppDelegate.swift project not found, skipping @main migration.');
      return;
```

Reply:
Added